### PR TITLE
Softap provisioning refactor

### PR DESCRIPTION
**Pull request by dachalco**
Description
-----------
Accommodate changes from PR #3187 for SoftAP.

Checklist:
----------

- [X] I have tested my changes. No regression in existing tests.
- [X] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: vendors/espressif/boards/ports/wifi/iot_softap_wifi_provisioning.c [hunk 1]
```diff
@@ -287,7 +377,7 @@ uint32_t IotWifiSoftAPProv_GetNumNetworks( void )
     return i;
 }
 
-uint32_t IotWifiSoftAPProv_Connect( uint32_t ulNetworkIndex )
+bool IotWifiSoftAPProv_Connect( uint32_t ulNetworkIndex )
```
Comment (ravibhagavandas) [inline vendors/espressif/boards/ports/wifi/iot_softap_wifi_provisioning.c:380]: Function return type changed to bool but returns `uint32_t` value.

File: vendors/espressif/boards/ports/wifi/iot_softap_wifi_provisioning.c [hunk 2]
```diff
@@ -265,11 +292,74 @@ uint32_t IotWifiSoftAPProv_Init( void )
...
+/* -------------------------------------------------------------------- */
+bool IotWifiSoftAPProv_Init( void )
+{
+    bool ret = true;
+
+    /* Reinit any cached credentials */
+    memset( &wifiProvisioning.xProvisionedParams, 0u, sizeof( wifiProvisioning.xProvisionedParams ) );
+    wifiProvisioning.xProvisionedParams.xSecurity = eWiFiSecurityWPA2;
+
+    wifiProvisioning.mutex = xSemaphoreCreateMutex();
```
Comment (ravibhagavandas) [inline vendors/espressif/boards/ports/wifi/iot_softap_wifi_provisioning.c:311]: Is the mutex used anywhere ?

File: vendors/espressif/boards/ports/wifi/iot_softap_wifi_provisioning.c [hunk 3]
```diff
@@ -265,11 +292,74 @@ uint32_t IotWifiSoftAPProv_Init( void )
...
+
+    if( wifiProvisioning.mutex == NULL )
+    {
+        ESP_LOGE( TAG, "Failed to create mutex for WiFi provisioning task." );
+        ret = false;
+    }
+
+    if( ret == true )
+    {
+        wifiProvisioning.messageQueue = xQueueCreate( 2U, sizeof( IotWifiSoftAPProvMessage_t ) );
```
Comment (ravibhagavandas) [inline vendors/espressif/boards/ports/wifi/iot_softap_wifi_provisioning.c:321]: could you define the queue size upfront ?

**Review by ravibhagavandas — APPROVED** [review]

**Review by paulbartell — APPROVED** [review]

**Pull request merged**